lib/core/src/Cardano/Wallet/Primitive/Types/UTxOIndex/Internal.hs

```diff
@@ -416,6 +420,9 @@ data InvariantStatus
     | InvariantAssetsInconsistent
       -- ^ Indicates that the 'index' and the cached 'balance' value disagree
       --   about which assets are included.
+    | InvariantIndexedCoinsAreIndeedCoins
+      -- ^ Indicates that all inputs indexed as 'coins' are indeed ada-only
+      -- outputs.
```

Comment:
This type represents all the possible invariant **failures**, rather than successes.

So we should almost certainly rename this constructor (and change the comment) to avoid confusion.

```suggestion
    | InvariantCoinIndexContainsNonCoins
      -- ^ Indicates that at least one entry in 'coins' is not ada-only.
```